macOS/DuckDuckGo/NewTabPage/NewTabPageSectionsAvailabilityProvider.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  NewTabPageSectionsAvailabilityProvider.swift
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import NewTabPage
+import BrowserServicesKit
+import Combine
+import FeatureFlags
+
+extension Notification.Name {
+
+    static var newTabPageSectionsAvailabilityDidChange = Notification.Name(rawValue: "newTabPageSectionsAvailabilityDidChange")
+
+}
+
+final class NewTabPageSectionsAvailabilityProvider: NewTabPageSectionsAvailabilityProviding {
+
+    private let featureFlagger: FeatureFlagger
+    private var cancellables = Set<AnyCancellable>()
+
+    internal init(featureFlagger: any FeatureFlagger) {
+        self.featureFlagger = featureFlagger
+
+        subscribeToOmnibarFeatureFlagChanges()
+    }
+
+    private func subscribeToOmnibarFeatureFlagChanges() {
+        guard let overridesHandler = featureFlagger.localOverrides?.actionHandler as? FeatureFlagOverridesPublishingHandler<FeatureFlag> else {
+            return
+        }
+
+        overridesHandler.flagDidChangePublisher
+            .filter { $0.0 == .newTabPageOmnibar }
+            .sink { _ in
+                NotificationCenter.default.post(name: .newTabPageSectionsAvailabilityDidChange, object: nil)
+            }
+            .store(in: &cancellables)
+    }
```

Comment:
This would only work for feature flag overrides (not for privacy config updates), hence it would only help internal users. I think we should be fine skipping the subscription and the notification (also in `AppearancePreferences`) – for production release, it will be completely OK to only apply the new flag state on new window creation (that’s how the HTML NTP feature flag was rolled out, for instance). So we can have simpler code and just ask internal users to reopen a browser window or restart the app.

Reply:
_Cross-posting my answer from Asana for future reference:_
If there is a percentage of users, who have the window open for ages, they won’t get the search feature until they restart the window, which is not ideal. Let’s keep the code for internal users there for now, we can remove it once the feature flag becomes global. If there is enough time, perhaps we could think about how to notify UI layer when the remote feature flag is changed